fallback.c

```diff
@@ -612,18 +612,45 @@ add_to_boot_list(CHAR16 *dirname, CHAR16 *filename, CHAR16 *label, CHAR16 *argum
 	if (EFI_ERROR(efi_status)) {
 		add_boot_option(dp, full_device_path, fullpath, label,
 				arguments);
-	} else if (option != 0) {
+	} else {
```

Comment:
We can save a level of indentation here by renaming `err:` to `done:` (or so) and then `goto done;` on the previous line (i.e., after add_boot_option()).

Reply:
Right, renamed to "done" and fixed the broken "return" stanza on error which was leaking memory in the mean time.